src/did/src/rpc/response.rs

```diff
@@ -0,0 +1,411 @@
+use core::fmt;
+use std::marker::PhantomData;
+
+use alloy::rpc::json_rpc::Response;
+use serde::de::{self, MapAccess, SeqAccess, Visitor};
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_bytes::ByteBuf;
+
+/// Response to a `RpcRequest`. If the request was a `Batch` the response will be a `Batch` as well and vice-versa for `One`
+#[derive(Clone, Debug, Serialize)]
+#[serde(untagged)]
+pub enum RpcResponse {
+    Single(Response),
+    Batch(Vec<Response>),
+}
+
+impl From<RpcResponse> for ByteBuf {
+    fn from(value: RpcResponse) -> Self {
+        match serde_json::to_vec(&value)
+            .map_err(|e| ByteBuf::from(e.to_string().as_bytes()))
+            .map(ByteBuf::from)
+        {
+            Ok(bytes) => bytes,
+            Err(e) => e,
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for RpcResponse {
```

Comment:
Why do we need a custom deserializer? We already had multiple issues where we used Custom deserializer but plan serializer